pulp_ansible/app/tasks/collections.py

```diff
@@ -303,50 +294,55 @@ def import_collection(
 def create_collection_from_importer(importer_result, metadata_only=False):
     """
     Process results from importer.
+
+    If ``metadata_only=True``, then this is being called in a sync pipeline, do not perform
+    any database operations, just return an unsaved CollectionVersion.
     """
     collection_info = importer_result["metadata"]
+    tags = collection_info.pop("tags")
 
-    with transaction.atomic():
-        collection, created = Collection.objects.get_or_create(
-            namespace=collection_info["namespace"], name=collection_info["name"]
-        )
-
-        tags = collection_info.pop("tags")
+    # Remove fields not used by this model
+    collection_info.pop("license_file")
+    collection_info.pop("readme")
 
-        # Remove fields not used by this model
-        collection_info.pop("license_file")
-        collection_info.pop("readme")
+    # the importer returns many None values. We need to let the defaults in the model prevail
+    for key in ["description", "documentation", "homepage", "issues", "repository"]:
+        if collection_info[key] is None:
+            collection_info.pop(key)
 
-        # the importer returns many None values. We need to let the defaults in the model prevail
-        for key in ["description", "documentation", "homepage", "issues", "repository"]:
-            if collection_info[key] is None:
-                collection_info.pop(key)
+    collection, created = Collection.objects.get_or_create(
```

Comment:
Isn't this a database operation? This breaks your comment about `metadata_only=True`.